library/core/src/iter/adapters/enumerate.rs

```diff
@@ -4,7 +4,7 @@ use crate::iter::adapters::{
 use crate::iter::{FusedIterator, InPlaceIterable, TrustedLen};
 use crate::ops::Try;
 
-/// An iterator that yields the current count and the element during iteration.
+/// An iterator that yields the zero-based index and the element during iteration.
```

Comment:
I don't think this improves much, but this just rephrases the current one. Looking at the other docs, we don't usually mention the index is zero-based, see an Iterator itself or `enumerate` method's docs.
Also, the method docs' example shows that the count is zero-based and that's enough to explain, I think.